include/ews/ews.hpp

```diff
@@ -5266,10 +5267,207 @@ namespace internal
 #else
     template <typename RequestHandler>
 #endif
-    inline std::string get_exchange_web_services_url(
-        const std::string& user_smtp_address, autodiscover_protocol protocol,
-        const basic_credentials& credentials, unsigned int redirections)
+    inline autodiscover_result get_exchange_web_services_url(
+        const std::string& user_smtp_address, const basic_credentials& credentials, unsigned int redirections,
+        const autodiscover_hints& hints)
     {
```

Comment:
Can you tell me why you copied the complete implementation to a different function? Now, you have two versions of `internal::get_exchange_web_services_url` with nearly identical code except the URL hinting part. Did you use Ctrl+C and Ctrl+V for this? Really?